src/App.js

```diff
@@ -49,4 +87,55 @@ function App() {
   );
 }
 
+function getOptions(answerValueSet) {
+  if(answerValueSet) {
+    return questionnaire.contained.find(e => `#${e.id}` === answerValueSet).compose.include[0].concept
+  }
+  return null
+}
+
+function OptionsList({ linkId, answerOptions, currentAnswer, dispatcher }) {
+  return answerOptions.map(concept => (
+    <div key={`${linkId}_${concept.code}`} className="form-check">
+    <input className="form-check-input"
+      type="radio" name={linkId}
+      id={`${linkId}_${concept.code}`}
+      value={concept.code}
+      checked={currentAnswer === concept.code}
+      onChange={() => dispatcher(concept.code)}
+      />
+    <label className="form-check-label" htmlFor={`${linkId}_${concept.code}`}>
+      {concept.display}
+    </label>
+  </div>
+  ))
+}
+
+// TODO: Input validation
+function URLInput({ linkId, currentAnswer, dispatcher}) {
+  return <input className="form-control" type="url" name={linkId} id={linkId} placeholder="https://example.com"
+              pattern="https?://.*" size="30"
+              required value={currentAnswer || ''} onChange={e => dispatcher(e.target.value)}/>
+}
```

Comment:
Perhaps you need to wrap these in a `<form>` to make use of the built-in validation capabilities. One problem with that is that you will have to have a submit button instead of download link looking like a button. However, in future versions we may have to have some kind of preview popup with a download button in it and such popup could be opened on submit...